spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/GlobalKTableBinder.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binder.kafka.streams;
+
+import org.apache.kafka.streams.kstream.GlobalKTable;
+
+import org.springframework.cloud.stream.binder.AbstractBinder;
+import org.springframework.cloud.stream.binder.Binding;
+import org.springframework.cloud.stream.binder.DefaultBinding;
+import org.springframework.cloud.stream.binder.ExtendedConsumerProperties;
+import org.springframework.cloud.stream.binder.ExtendedProducerProperties;
+import org.springframework.cloud.stream.binder.ExtendedPropertiesBinder;
+import org.springframework.cloud.stream.binder.kafka.provisioning.KafkaTopicProvisioner;
+import org.springframework.cloud.stream.binder.kafka.streams.properties.KafkaStreamsBinderConfigurationProperties;
+import org.springframework.cloud.stream.binder.kafka.streams.properties.KafkaStreamsConsumerProperties;
+import org.springframework.cloud.stream.binder.kafka.streams.properties.KafkaStreamsExtendedBindingProperties;
+import org.springframework.cloud.stream.binder.kafka.streams.properties.KafkaStreamsProducerProperties;
+import org.springframework.util.StringUtils;
+
+/**
+ * {@link org.springframework.cloud.stream.binder.Binder} implementation for {@link GlobalKTable}.
+ * This implementation extends from the {@link AbstractBinder} directly.
+ *
+ * Provides only consumer binding for the bound GlobalKTable as output bindings are not allowed on it.
+ *
+ * @author Soby Chacko
+ */
+public class GlobalKTableBinder extends
+		AbstractBinder<GlobalKTable<Object, Object>, ExtendedConsumerProperties<KafkaStreamsConsumerProperties>, ExtendedProducerProperties<KafkaStreamsProducerProperties>>
```

Comment:
maybe we can provide this as an enhancement later?

Reply:
Huh? Why later?
This is a feature of this new Binder and what we have so far is confusing.
What is the point to commit wrong code and even release it when we already know that it is wrong and we see a solution.

What is the point of review then if we do nothing? 🤷‍♂️ 

Sorry, if I'm so nasty, but I don't think this is the way Spring evolve...

Reply:
We throw `UnsupportedOperationException` from the producer binding method. So, anyone who is trying to bind an output will get that exception anyway. I don't see the necessity for doing that here. Maybe I am missing something?

Reply:
Right, but we don't throw it from the `KafkaStreamsProducerProperties`. So, I will spend some time (a lot of time in case of YAML!) to configure producer properties just to fail in the end.
If we would have some blocker just from the API level, there is a little chance that people would try to configure producer properties.
Also some special no-op type wouldn't let us to fail with our internal logic and we wouldn't have any implementation for producer behavior.

Anyway I might just don't see the whole picture and all this discussion may just lead to dead end just because we may not achieve the solution with Java (or YAML?) restrictions.

So, let it really be and let's try to revise it later!